src/System.Reflection.Metadata/tests/Metadata/Decoding/MethodSignatureTests.cs

```diff
@@ -26,5 +29,60 @@ public void VerifyConstructor()
             Assert.Equal(methodSignature.GenericParameterCount, genericParameterCount);
             Assert.Equal(methodSignature.ParameterTypes, parameterTypes);
         }
+
+        [Fact]
+        public unsafe void VerifyMultipleOptionalModifiers()
+        {
+            // Type 1: int32 modopt([mscorlib]System.Runtime.CompilerServices.IsLong) modopt([mscorlib]System.Runtime.CompilerServices.CallConvCdecl)
+            // Type 2: char* 
+            // Type 3: uint32 
+            // Type 4: char modopt([mscorlib]System.Runtime.CompilerServices.IsConst)*
+            var testSignature = new byte[] { 0x20, 0x45, 0x20, 0x69, 0x08, 0x0F, 0x03, 0x09, 0x0F, 0x20, 0x55, 0x03 };
+            var types = new string[]
+            {
+                "11 1A Int32",  // This has multiple modifiers, which exercises different code paths than single modifiers
+                "Char*",
+                "UInt32",
+                "15 Char*"
+            };
+
+            fixed (byte* testSignaturePtr = &testSignature[0])
+            {
+                var signatureBlob = new BlobReader(testSignaturePtr, testSignature.Length);
+                var sigTypeProvider = new TestSignatureTypeProvider();
+
+                foreach (string typeString in types)
+                {
+                    // Verify that each type is decoded as expected
+                    Assert.Equal(typeString, SignatureDecoder.DecodeType(ref signatureBlob, new TestSignatureTypeProvider()));
```

Comment:
Why create another instance of TestSignatureTypeProvider here?  You created another one above and it doesn't appear to track any state.